src/components/AddTransactionForm/AddTransactionForm.tsx

```diff
@@ -21,27 +23,40 @@ export type TransactionTableData = z.infer<typeof transactionTableDataSchema>;
 type AddTransactionFormProps = {
   onClose: () => void;
   onReopen: () => void;
+  transactionToEdit?: TransactionTableData;
 };
 
 export const AddTransactionForm = ({
   onClose,
   onReopen,
+  transactionToEdit,
 }: AddTransactionFormProps) => {
   const { data: symbolList, isLoading } = useSymbolList();
+  const addTransaction = useTransactionStore((state) => state.addTransaction);
+  const editTransaction = useTransactionStore((state) => state.editTransaction);
 
   const methods = useForm<AddTransactionFormFields>({
     resolver: zodResolver(formFieldsSchema),
+    defaultValues: transactionToEdit
+      ? ({
+          symbol: transactionToEdit.holding.holdingSymbol,
+          transactionType: transactionToEdit.transactionType,
+          date: transactionToEdit.transactionDate,
+          name: transactionToEdit.holding.holdingName,
+          quantity: transactionToEdit.numberOfStocks,
+          price: transactionToEdit.transactionValue.perShare,
+          currency: transactionToEdit.transactionValue.currency,
+          fee: transactionToEdit.transactionFee?.total || 0,
+        } as AddTransactionFormFields)
+      : undefined,
   });
 
   const onSubmit: SubmitHandler<AddTransactionFormFields> = (
     data: AddTransactionFormFields,
   ) => {
     try {
-      const existingTransactions = JSON.parse(
-        localStorage.getItem('transactions') || '[]',
-      ) as TransactionTableData[];
-
       const transactionToSave: TransactionTableData = {
+        id: transactionToEdit?.id || v4(),
```

Comment:
You don't need to add `uuid`, javascript has it built in within `crypto`. Additionally you should use `??` as `||` can give you unexpected results in special cases.
```suggestion
        id: transactionToEdit?.id ?? crypto.randomUUID(),
```